test/unit/math/rev/functor/test_fixture_dae_chem.hpp

```diff
@@ -0,0 +1,398 @@
+                     "DAE parameters and data");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_nan.str());
+    this->x_r = x_r_;
+
+    // inf test
+    std::stringstream expected_is_inf;
+    expected_is_inf << "is " << std::numeric_limits<double>::infinity();
+    std::stringstream expected_is_neg_inf;
+    expected_is_neg_inf << "is " << -std::numeric_limits<double>::infinity();
+    double inf = std::numeric_limits<double>::infinity();
+
+    this->yy0[0] = inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error, "initial state");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_inf.str());
+    this->yy0 = yy0_;
+
+    this->yp0[0] = -inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     "initial state derivative");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_neg_inf.str());
+    this->yp0 = yp0_;
+
+    this->t0 = inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error, "initial time");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_inf.str());
+    this->t0 = -inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error, "initial time");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_neg_inf.str());
+    this->t0 = t0_;
+
+    this->ts.back() = inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error, "times");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_inf.str());
+    this->ts.back() = -inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error, "times");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_neg_inf.str());
+    this->ts = ts_;
+
+    this->theta[0] = inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     "DAE parameters and data");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_inf.str());
+    this->theta[0] = -inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     "DAE parameters and data");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_neg_inf.str());
+    this->theta = theta_;
+
+    this->x_r = std::vector<double>{inf};
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     "DAE parameters and data");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_inf.str());
+    this->x_r[0] = -inf;
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     "DAE parameters and data");
+    EXPECT_THROW_MSG(apply_solver_tol(), std::domain_error,
+                     expected_is_neg_inf.str());
+    this->x_r = x_r_;
+  }
```

Comment:
We probably need to test the error messages which are returned when there is an IDAS failure (i.e. translation of flags into readable error messages). Has this been done for the ODE solvers?

Reply:
Yes. ODE solvers use similar error tests, such as here https://github.com/stan-dev/math/blob/ff7815b238dc1c026730d373e9fe8988ec4dae21/test/unit/math/rev/functor/test_fixture_ode_sho.hpp#L107